### PR TITLE
Set up CI build for CUDA 9.2 + macOS

### PR DESCRIPTION
**Pull request by yf225**
CI build for CUDA 9.2 + macOS will be enabled after this PR is merged.

### CODE REVIEW

**Comment by bddppq** [comment]
I'm just curious, what's the purpose of compiling cuda on osx? Is there a way to hook up an nvidia card in decent osx machines? :-)

**Comment by ezyang** [comment]
We have a number of (reasonably vocal) users who use CUDA on OSX, so, yes!

File: aten/src/ATen/native/cuda/TensorFactories.cu [hunk 1]
```diff
@@ -43,14 +43,8 @@ Tensor& randperm_out_cuda(Tensor& result, int64_t n, Generator* generator) {
     throw std::runtime_error(oss.str());
   }
 
-  if (n > 0) {
-    AT_DISPATCH_ALL_TYPES_AND_HALF(
-      result.type(), "randperm_out_cuda", [&] {
-        AT_CHECK(Scalar(n).to<scalar_t>(),
-          "n is too large for result tensor type: '", result.type().toString(), "'");
```
Comment (ezyang) [inline aten/src/ATen/native/cuda/TensorFactories.cu]: What's going on with this diff?

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/native/cuda/TensorFactories.cu (hunk 1 above)
Comment (yf225) [reply aten/src/ATen/native/cuda/TensorFactories.cu]: The old code will give 
```
Undefined symbols for architecture x86_64:
  "std::__1::enable_if<!(::std::__1::is_integral<double>::value), bool>::type at::overflows<at::Half, double>(double)", referenced from:
      at::Half at::checked_convert<at::Half, double>(double, char const*) in caffe2_gpu_generated_TensorFactories.cu.o
  "std::__1::enable_if<::std::__1::is_integral<long long>::value, bool>::type at::overflows<at::Half, long long>(long long)", referenced from:
      at::Half at::checked_convert<at::Half, long long>(long long, char const*) in caffe2_gpu_generated_TensorFactories.cu.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
when compiling using NVCC in macOS. I haven't figured out why it happens, but I think the new code is cleaner anyway.

**Pull request merged**